usr.sbin/adduser/adduser.sh

```diff
@@ -613,6 +654,84 @@ get_password() {
 	fi
 }
 
+# get_zfs_encryption
+#	Read ZFS options for the user's home dataset.
+#
+get_zfs_encryption() {
+	_input=
+	_prompt="Enable ZFS encryption? (yes/no) [no]: "
+	while : ; do
+		echo -n "$_prompt"
+		read _input
+
+		[ -z "$_input" ] && _input=Zencrypt
+		case $_input in
+		[Nn][Oo]|[Nn])
+			Zencrypt=
+			break
+			;;
+		[Yy][Ee][Ss]|[Yy][Ee]|[Yy])
+			Zencrypt=yes
+			break
+			;;
+		*)
+			# invalid answer; repeat loop
+			continue
+			;;
+		esac
+	done
+
+	if [ -n "${Zencrypt}" ]; then
+		zfsopt="-o encryption=on -o keylocation=prompt -o keyformat=passphrase"
+	fi
+}
+
+# create_zfs_chrooted_dataset
+#   Create ZFS dataset owned by the user that was just added within a bsdinstall chroot
+#
+create_zfs_chrooted_dataset() {
+	${ZFSCMD} create -u ${zfsopt} "${zhome}"
+	if [ "$?" -ne 0 ]; then
+		err "There was an error creating ZFS dataset (${zhome})."
+		return 1
+	fi
+	${ZFSCMD} set mountpoint=legacy "${zhome}"
+	${MKDIRCMD} -p "${uhome}"
+	${MOUNTCMD} -t zfs "${zhome}" "${uhome}"
+}
+
+# umount_legacy_zfs
+#   Unmount ZFS home directory created as a legacy mount and switch inheritance
+#
+umount_legacy_zfs() {
+	${UMOUNTCMD} "${uhome}"
+	${ZFSCMD} inherit mountpoint "${zhome}"
+}
+
+# create_zfs_dataset
+#   Create ZFS dataset owned by the user that was just added.
+#
+create_zfs_dataset() {
+	${ZFSCMD} create ${zfsopt} "${zhome}"
```

Comment:
Ditto above; you're not using the exit code otherwise, might as well just test it directly